include/brightness.h

```diff
@@ -0,0 +1,35 @@
+#ifndef POKEHEARTGOLD_BRIGHTNESS_H
+#define POKEHEARTGOLD_BRIGHTNESS_H
+
+#include "global.h"
+#include <nitro/gx/g2.h>
+
+#define SCREEN_MASK_MAIN 1
+#define SCREEN_MASK_SUB 2
+
+typedef struct BrightnessData {
+    GXBlendPlaneMask surfaceMask;
+    u32 screenMask;
+    u16 stepCount;
+    s16 targetBrightness;
+    u32 unk0c;
+    s8 transitionDirection;
+    s16 brightnessDiff;
+    s16 stepSizeInteger;
+    u16 stepSizeFractional;
+    u16 fractionalCount;
+    fx16 currentBrightness;
+    BOOL transitionActive;
+} BrightnessData;
+
+void DoBrightnessTransitionStep(BrightnessData *brightness);
```

Comment:
We could have a Brightness_ prefix for all these functions, not really required but IIRC that's what the style is supposed to be

Reply:
I'd rather not tbh, since it's pretty clear what these funcs belong to